torch/csrc/jit/backends/backend_detail.cpp

```diff
@@ -58,10 +58,11 @@ Module codegen_backend_module(
   // Clone orig_module to make sure backend transformation is
   // functional.
   auto cloned_module = orig_module.clone();
+  auto module_name = orig_module.type()->name()->qualifiedName();
 
   // Generate LoweredModule.
   Module loweredModule(
-      "torch.jit." + backend_name + "LoweredModule",
+      "torch.jit." + backend_name + "_" + module_name + "_LoweredModule",
```

Comment:
So are we sure that two instances of the same module type will not get lowered separately? What happens if that is the case? We will run into the same issue, no? I think we should probably use some unique id as well.

Reply:
Yeah, I think we also discussed this back then (the need some unique ID).

Reply:
I think the qualified name of a module is unique. @suo and @SplitInfinity , could you confirm?

Reply:
I dont think thats the case. You can instantiate one module type with many instances.

Reply:
I think what @kimishpatel means is that you can have a parent module with 2 submodules of the same type.
So the submodule will indeed have a unique name, but then it will show up twice in the parent module.

If that has a problem then we need to mangle another unique id; otherwise, then what you did is enough.

Reply:
Let me quickly confirm "a parent module with 2 submodules of the same type" case and see if they have the same qualified name. Thanks @raziel and @kimishpatel !

Reply:
After diving deeper, there are actually two issues:
1. The issue for delegation that this PR is to resolve. Basically, a new Module is constructed here, with a class name as the first argument of the constructor. This class name should reflect the original class name. After this fix, different classes lowered to the same backend would also have different class names. 
2. General issue of bytecode serialization. For the same original class name but different instances (@kimishpatel and @raziel raised here). It is handled in TorchScript serialization, by mangling the names in a TypeNameUniquer. However, it's not mangled when serializing bytecode. As a result, there is discrepancy between the names in TS and in bytecode. This discrepancy appears only once at the first serialization. When it's loaded again the mangled name would be taken from bytecode. T93782563 is created to follow up this issue.

So for issue #1 we need to resolve it anyway I think this PR is still valid for that. The unique id for an instance is a separate issue and can be addressed in a centralized place in bytecode serialization.

Reply:
Thanks. Sounds good.

Reply:
> So this is a general issue in TS that potentially affects any custom class?

I think so. Not only for custom classes but for all class types. TS has handled it using `type_name_uniquer_` in `ScriptModuleSerializer`. I think bytecode needs to reflect the same pattern.